### PR TITLE
Fogl 653.d

### PR DESCRIPTION
**Pull request by praveen-garg**
a PR on top of FOGL-653 #451 
1) allow shutdown via `http://localhost:<core port>/foglamp/service/shutdown` POST only; 
2) added TODO for scheduler timeout and let syslog reveal the problems

> everything stops with shutdown call except north plugins tasks cleanup i.e. FOGL-820

### CODE REVIEW

File: python/foglamp/services/core/server.py [hunk 1]
```diff
@@ -419,14 +419,14 @@ def _start_core(cls, loop=None):
...
-            loop.close()
+            # print("(Press CTRL+C to quit)")
+            # try:
+            loop.run_forever()
+            # except KeyboardInterrupt:
+            #     pass
+            # finally:
+            #     cls.stop(loop)
+            # loop.close()
         except (OSError, RuntimeError, TimeoutError) as e:
```
Comment (praveen-garg) [inline python/foglamp/services/core/server.py]: @MarkRiddoch Please let me know, we want to remove this ctrl+c ? as we want to start with scripts only.

File: python/foglamp/services/core/server.py [hunk 2]
```diff
@@ -569,7 +559,12 @@ def stop(cls, loop=None):
             cls.scheduler = None
         except TimeoutError as e:
             _logger.exception('Unable to stop the scheduler')
-            raise e
+            # TODO: FOGL-xxx Scheduler gets timeout,
```
Comment (praveen-garg) [inline python/foglamp/services/core/server.py]: if we decide to merge in, I shall add a story with related tasks / bugs

File: python/foglamp/services/south/server.py [hunk 3]
```diff
@@ -199,9 +199,10 @@ def run(self):
 
         try:
             await Ingest.stop()
+            _LOGGER.exception('Stopped the Ingest server.')
         except Exception as ex:
             _LOGGER.exception('Unable to stop the Ingest server. %s', str(ex))
-            return
+            # return
```
Comment (praveen-garg) [inline python/foglamp/services/south/server.py]: Ingest must stop in its own subprocess, south POLL plugin starts (and does not stop) it even when its get timeout and disconnected

**Comment by ashish-jabble** [comment]
Below are my findings for south plugin shutdown against this branch

1) **COAP** - No regression 

2) **HTTP_SOUTH** - default port: 6683 
When I change configuration value for port (6684), callback triggers and got below error in plugin_shutdown
`curl -X PUT -d '{"value":"6684"}' http://localhost:8081/foglamp/category/HTTP_SOUTH/port | jq`

```
Jan 24 13:26:48 nerd-034 FogLAMP[5244] ERROR: http_south: foglamp.plugins.south.http_south.http_south: 'Task' object has no attribute 'close'#012Traceback (most recent call last):#012  File "/home/foglamp/Development/FogLAMP/python/foglamp/plugins/south/http_south/http_south.py", line 120, in plugin_shutdown#012    server.close()#012AttributeError: 'Task' object has no attribute 'close'
```

3) **CC2650POLL** - default pollinterval: 500
When I change configuration value for pollinterval (i.e 600), callback triggers and got below error in plugin_shutdown
`curl -X PUT -d '{"value":"600"}' http://localhost:8081/foglamp/category/CC2650POLL/pollInterval | jq`
```
Jan 24 13:10:14 nerd-034 FogLAMP[4638] INFO: cc2650poll: foglamp.plugins.south.cc2650poll.cc2650poll: SensorTagCC2650 24:71:89:CF:AE:05 Disconnected.
Jan 24 13:10:14 nerd-034 FogLAMP[4638] INFO: cc2650poll: foglamp.plugins.south.cc2650poll.cc2650poll: CC2650 poll plugin shut down.
Jan 24 13:10:14 nerd-034 FogLAMP[4638] INFO: cc2650poll: foglamp.plugins.south.cc2650poll.cc2650poll: Old config for CC2650POLL plugin {'shutdownThreshold': {'description': 'Time in seconds allowed for shutdown to complete the pending tasks', 'value': '10', 'type': 'integer', 'default': '10'}, 'pollInterval': {'description': 'The interval between poll calls to the South device poll routine expressed in milliseconds.', 'value': '600', 'type': 'integer', 'default': '500'}, 'plugin': {'description': 'Sensortag CC2650 poll type plugin', 'value': 'cc2650poll', 'type': 'string', 'default': 'cc2650poll'}, 'tag': <foglamp.plugins.south.common.sensortag_cc2650.SensorTagCC2650 object at 0x7fd4831e06a0>, 'characteristics': {'pressure': {'configuration': {'uuid': 'f000aa42-0451-4000-b000-000000000000', 'handle': '0x0037'}, 'period': {'uuid': 'f000aa43-0451-4000-b000-000000000000', 'handle': '0x0000'}, 'data': {'uuid': 'f000aa41-0451-4000-b000-000000000000', 'handle': '0x0034'}}, 'temperature': {'configuration': {'uuid': 'f000aa02-0451-4000-b000-000000000000', 'handle': '0x0027'}, 'period': {'uuid': 'f000aa03-0451-4000-b000-000000000000', 'handle': '0x0029'}, 'data': {'uuid': 'f000aa01-0451-4000-b000-000000000000', 'handle': '0x0024'}}, 'humidity': {'configuration': {'uuid': 'f000aa22-0451-4000-b000-000000000000', 'handle': '0x002f'}, 'period': {'uuid': 'f000aa23-0451-4000-b000-000000000000', 'handle': '0x0031'}, 'data': {'uuid': 'f000aa21-0451-4000-b000-000000000000', 'handle': '0x002c'}}, 'battery': {'data': {'uuid': '00002a19-0000-1000-8000-00805f9b34fb', 'handle': '0x001e'}}, 'luminance': {'configuration': {'uuid': 'f000aa72-0451-4000-b000-000000000000', 'handle': '0x0047'}, 'period': {'uuid': 'f000aa73-0451-4000-b000-000000000000', 'handle': '0x0049'}, 'data': {'uuid': 'f000aa71-0451-4000-b000-000000000000', 'handle': '0x0044'}}, 'movement': {'configuration': {'uuid': 'f000aa82-0451-4000-b000-000000000000', 'handle': '0x003f'}, 'period': {'uuid': 'f000aa83-0451-4000-b000-000000000000', 'handle': '0x0041'}, 'data': {'uuid': 'f000aa81-0451-4000-b000-000000000000', 'handle': '0x003c'}}}, 'is_connected': True, 'bluetoothAddress': {'description': 'Bluetooth MAC address', 'value': '24:71:89:CF:AE:05', 'type': 'string', 'default': '24:71:89:CF:AE:05'}, 'connectionTimeout': {'description': 'BLE South Device timeout value in seconds', 'value': '10', 'type': 'integer', 'default': '10'}} #012 new config {'shutdownThreshold': {'description': 'Time in seconds allowed for shutdown to complete the pending tasks', 'value': '10', 'type': 'integer', 'default': '10'}, 'pollInterval': {'description': 'The interval between poll calls to the South device poll routine expressed in milliseconds.', 'value': '100', 'type': 'integer', 'default': '500'}, 'plugin': {'description': 'Sensortag CC2650 poll type plugin', 'value': 'cc2650poll', 'type': 'string', 'default': 'cc2650poll'}, 'bluetoothAddress': {'description': 'Bluetooth MAC address', 'value': '24:71:89:CF:AE:05', 'type': 'string', 'default': '24:71:89:CF:AE:05'}, 'connectionTimeout': {'description': 'BLE South Device timeout value in seconds', 'value': '10', 'type': 'integer', 'default': '10'}}
Jan 24 13:10:17 nerd-034 FogLAMP[4638] ERROR: sensortag_cc2650: foglamp.plugins.south.common.sensortag_cc2650: SensorTagCC2650 24:71:89:CF:AE:05 retrying fetching characteristics...#012Traceback (most recent call last):#012  File "/home/foglamp/.local/lib/python3.5/site-packages/pexpect/expect.py", line 97, in expect_loop#012    incoming = spawn.read_nonblocking(spawn.maxread, timeout)#012  File "/home/foglamp/.local/lib/python3.5/site-packages/pexpect/pty_spawn.py", line 452, in read_nonblocking#012    raise TIMEOUT('Timeout exceeded.')#012pexpect.exceptions.TIMEOUT: Timeout exceeded.#012#012During handling of the above exception, another exception occurred:#012#012Traceback (most recent call last):#012  File "/home/foglamp/Development/FogLAMP/python/foglamp/plugins/south/common/sensortag_cc2650.py", line 186, in get_char_handle#012    self.con.expect('.*handle:.* \r', timeout=3)#012  File "/home/foglamp/.local/lib/python3.5/site-packages/pexpect/spawnbase.py", line 315, in expect#012    timeout, searchwindowsize, async)#012  File "/home/foglamp/.local/lib/python3.5/site-packages/pexpect/spawnbase.py", line 339, in expect_list#012    return exp.expect_loop(timeout)#012  File "/home/foglamp/.local/lib/python3.5/site-packages/pexpect/expect.py", line 104, in expect_loop#012    return self.timeout(e)#012  File "/home/foglamp/.local/lib/python3.5/site-packages/pexpect/expect.py", line 68, in timeout#012    raise TIMEOUT(msg)#012pexpect.exceptions.TIMEOUT: Timeout exceeded.#012<pexpect.pty_spawn.spawn object at 0x7fd48243a240>#012command: /usr/bin/gatttool#012args: ['/usr/bin/gatttool', '-b', '24:71:89:CF:AE:05', '--interactive']#012searcher: None#012buffer (last 100 chars): b'tics by UUID failed: No attribute found within the given range\r\n\x1b[0;94m[24:71:89:CF:AE:05]\x1b[0m[LE]> '#012before (last 100 chars): b'tics by UUID failed: No attribute found within the given range\r\n\x1b[0;94m[24:71:89:CF:AE:05]\x1b[0m[LE]> '#012after: <class 'pexpect.exceptions.TIMEOUT'>#012match: None#012match_index: None#012exitstatus: None#012flag_eof: False#012pid: 4727#012child_fd: 25#012closed: False#012timeout: 30#012delimiter: <class 'pexpect.exceptions.EOF'>#012logfile: None#012logfile_read: None#012logfile_send: None#012maxread: 2000#012ignorecase: False#012searchwindowsize: None#012delaybeforesend: 0.05#012delayafterclose: 0.1#012delayafterterminate: 0.1
Jan 24 13:10:19 nerd-034 FogLAMP[4638] ERROR: server: foglamp.services.south.server: Failed to poll for plugin CC2650POLL, retry count: 2#012Traceback (most recent call last):#012  File "/home/foglamp/Development/FogLAMP/python/foglamp/services/south/server.py", line 164, in _exec_plugin_poll#012    await asyncio.sleep(sleep_seconds)#012  File "/usr/lib/python3.5/asyncio/tasks.py", line 516, in sleep#012    return (yield from future)#012  File "/usr/lib/python3.5/asyncio/futures.py", line 361, in __iter__#012    yield self  # This tells Task to wait for completion.#012  File "/usr/lib/python3.5/asyncio/tasks.py", line 296, in _wakeup#012    future.result()#012  File "/usr/lib/python3.5/asyncio/futures.py", line 266, in result#012    raise CancelledError#012concurrent.futures._base.CancelledError
Jan 24 13:10:21 nerd-034 FogLAMP[4638] ERROR: cc2650poll: foglamp.plugins.south.cc2650poll.cc2650poll: SensorTagCC2650 24:71:89:CF:AE:05 exception: #012Traceback (most recent call last):#012  File "/home/foglamp/Development/FogLAMP/python/foglamp/services/south/server.py", line 164, in _exec_plugin_poll#012    await asyncio.sleep(sleep_seconds)#012  File "/usr/lib/python3.5/asyncio/tasks.py", line 516, in sleep#012    return (yield from future)#012  File "/usr/lib/python3.5/asyncio/futures.py", line 361, in __iter__#012    yield self  # This tells Task to wait for completion.#012  File "/usr/lib/python3.5/asyncio/tasks.py", line 296, in _wakeup#012    future.result()#012  File "/usr/lib/python3.5/asyncio/futures.py", line 266, in result#012    raise CancelledError#012concurrent.futures._base.CancelledError#012#012During handling of the above exception, another exception occurred:#012#012Traceback (most recent call last):#012  File "/home/foglamp/Development/FogLAMP/python/foglamp/plugins/south/cc2650poll/cc2650poll.py", line 145, in plugin_poll#012    raise RuntimeError#012RuntimeError
Jan 24 13:10:21 nerd-034 FogLAMP[4638] ERROR: server: foglamp.services.south.server: Failed to poll for plugin CC2650POLL, retry count: 3#012Traceback (most recent call last):#012  File "/home/foglamp/Development/FogLAMP/python/foglamp/services/south/server.py", line 164, in _exec_plugin_poll#012    await asyncio.sleep(sleep_seconds)#012  File "/usr/lib/python3.5/asyncio/tasks.py", line 516, in sleep#012    return (yield from future)#012  File "/usr/lib/python3.5/asyncio/futures.py", line 361, in __iter__#012    yield self  # This tells Task to wait for completion.#012  File "/usr/lib/python3.5/asyncio/tasks.py", line 296, in _wakeup#012    future.result()#012  File "/usr/lib/python3.5/asyncio/futures.py", line 266, in result#012    raise CancelledError#012concurrent.futures._base.CancelledError#012#012During handling of the above exception, another exception occurred:#012#012Traceback (most recent call last):#012  File "/home/foglamp/Development/FogLAMP/python/foglamp/plugins/south/cc2650poll/cc2650poll.py", line 145, in plugin_poll#012    raise RuntimeError#012RuntimeError#012#012During handling of the above exception, another exception occurred:#012#012Traceback (most recent call last):#012  File "/home/foglamp/Development/FogLAMP/python/foglamp/services/south/server.py", line 149, in _exec_plugin_poll#012    data = self._plugin.plugin_poll(self._plugin_handle)#012  File "/home/foglamp/Development/FogLAMP/python/foglamp/plugins/south/cc2650poll/cc2650poll.py", line 246, in plugin_poll#012    raise exceptions.DataRetrievalError(ex)#012foglamp.services.south.exceptions.DataRetrievalError
Jan 24 13:10:23 nerd-034 FogLAMP[4638] ERROR: cc2650poll: foglamp.plugins.south.cc2650poll.cc2650poll: SensorTagCC2650 24:71:89:CF:AE:05 exception: #012Traceback (most recent call last):#012  File "/home/foglamp/Development/FogLAMP/python/foglamp/services/south/server.py", line 164, in _exec_plugin_poll#012    await asyncio.sleep(sleep_seconds)#012  File "/usr/lib/python3.5/asyncio/tasks.py", line 516, in sleep#012    return (yield from future)#012  File "/usr/lib/python3.5/asyncio/futures.py", line 361, in __iter__#012    yield self  # This tells Task to wait for completion.#012  File "/usr/lib/python3.5/asyncio/tasks.py", line 296, in _wakeup#012    future.result()#012  File "/usr/lib/python3.5/asyncio/futures.py", line 266, in result#012    raise CancelledError#012concurrent.futures._base.CancelledError#012#012During handling of the above exception, another exception occurred:#012#012Traceback (most recent call last):#012  File "/home/foglamp/Development/FogLAMP/python/foglamp/plugins/south/cc2650poll/cc2650poll.py", line 145, in plugin_poll#012    raise RuntimeError#012RuntimeError
Jan 24 13:10:23 nerd-034 FogLAMP[4638] ERROR: server: foglamp.services.south.server: Failed to poll for plugin CC2650POLL, retry count: 4#012Traceback (most recent call last):#012  File "/home/foglamp/Development/FogLAMP/python/foglamp/services/south/server.py", line 164, in _exec_plugin_poll#012    await asyncio.sleep(sleep_seconds)#012  File "/usr/lib/python3.5/asyncio/tasks.py", line 516, in sleep#012    return (yield from future)#012  File "/usr/lib/python3.5/asyncio/futures.py", line 361, in __iter__#012    yield self  # This tells Task to wait for completion.#012  File "/usr/lib/python3.5/asyncio/tasks.py", line 296, in _wakeup#012    future.result()#012  File "/usr/lib/python3.5/asyncio/futures.py", line 266, in result#012    raise CancelledError#012concurrent.futures._base.CancelledError#012#012During handling of the above exception, another exception occurred:#012#012Traceback (most recent call last):#012  File "/home/foglamp/Development/FogLAMP/python/foglamp/plugins/south/cc2650poll/cc2650poll.py", line 145, in plugin_poll#012    raise RuntimeError#012RuntimeError#012#012During handling of the above exception, another exception occurred:#012#012Traceback (most recent call last):#012  File "/home/foglamp/Development/FogLAMP/python/foglamp/services/south/server.py", line 149, in _exec_plugin_poll#012    data = self._plugin.plugin_poll(self._plugin_handle)#012  File "/home/foglamp/Development/FogLAMP/python/foglamp/plugins/south/cc2650poll/cc2650poll.py", line 246, in plugin_poll#012    raise exceptions.DataRetrievalError(ex)#012foglamp.services.south.exceptions.DataRetrievalError
```

4) **CC2650ASYN** - default connectionTimeout: 10
When I change configuration value for connectionTimeout (i.e 20), callback triggers and got below error in reconfiguration that's separate issue but **plugin shutdown is working fine
plugin shutdown is working fine** for this case

`curl -X PUT -d '{"value":"20"}' http://localhost:8081/foglamp/category/CC2650ASYN/connectionTimeout | jq`

```
Jan 24 13:14:13 nerd-034 FogLAMP[4897] ERROR: cc2650async: foglamp.plugins.south.cc2650async.cc2650async: SensorTagCC2650 async 24:71:89:CF:AE:05 exception: unhashable type: 'dict'#012Traceback (most recent call last):#012  File "/home/foglamp/Development/FogLAMP/python/foglamp/plugins/south/cc2650async/cc2650async.py", line 271, in save_data#012    'readings': {movement[reading_key]}#012TypeError: unhashable type: 'dict'
Jan 24 13:15:01 nerd-034 CRON[4985]: (root) CMD (command -v debian-sa1 > /dev/null && debian-sa1 1 1)
Jan 24 13:15:32 nerd-034 FogLAMP[4897] INFO: cc2650async: foglamp.plugins.south.cc2650async.cc2650async: SensorTagCC2650 (async) 24:71:89:CF:AE:05 Disconnected.
Jan 24 13:15:32 nerd-034 FogLAMP[4897] INFO: cc2650async: foglamp.plugins.south.cc2650async.cc2650async: CC2650 async plugin shut down.
Jan 24 13:15:32 nerd-034 FogLAMP[4897] INFO: cc2650async: foglamp.plugins.south.cc2650async.cc2650async: Old config for CC2650ASYN plugin {'bluetoothAddress': {'value': '24:71:89:CF:AE:05', 'type': 'string', 'description': 'Bluetooth MAC address', 'default': '24:71:89:CF:AE:05'}, 'notification_handles': ['0x001f', '0x0025', '0x002d', '0x0035', '0x003d'], 'tag': <foglamp.plugins.south.common.sensortag_cc2650.SensorTagCC2650 object at 0x7f1487ed3b38>, 'shutdownThreshold': {'value': '10', 'type': 'integer', 'description': 'Time in seconds allowed for shutdown to complete the pending tasks', 'default': '10'}, 'characteristics': {'temperature': {'configuration': {'uuid': 'f000aa02-0451-4000-b000-000000000000', 'handle': '0x0027'}, 'data': {'uuid': 'f000aa01-0451-4000-b000-000000000000', 'handle': '0x0024'}, 'period': {'uuid': 'f000aa03-0451-4000-b000-000000000000', 'handle': '0x0029'}}, 'luminance': {'configuration': {'uuid': 'f000aa72-0451-4000-b000-000000000000', 'handle': '0x0047'}, 'data': {'uuid': 'f000aa71-0451-4000-b000-000000000000', 'handle': '0x0044'}, 'period': {'uuid': 'f000aa73-0451-4000-b000-000000000000', 'handle': '0x0049'}}, 'pressure': {'configuration': {'uuid': 'f000aa42-0451-4000-b000-000000000000', 'handle': '0x0037'}, 'data': {'uuid': 'f000aa41-0451-4000-b000-000000000000', 'handle': '0x0034'}, 'period': {'uuid': 'f000aa43-0451-4000-b000-000000000000', 'handle': '0x0000'}}, 'battery': {'data': {'uuid': '00002a19-0000-1000-8000-00805f9b34fb', 'handle': '0x001e'}}, 'movement': {'configuration': {'uuid': 'f000aa82-0451-4000-b000-000000000000', 'handle': '0x003f'}, 'data': {'uuid': 'f000aa81-0451-4000-b000-000000000000', 'handle': '0x003c'}, 'period': {'uuid': 'f000aa83-0451-4000-b000-000000000000', 'handle': '0x0041'}}, 'humidity': {'configuration': {'uuid': 'f000aa22-0451-4000-b000-000000000000', 'handle': '0x002f'}, 'data': {'uuid': 'f000aa21-0451-4000-b000-000000000000', 'handle': '0x002c'}, 'period': {'uuid': 'f000aa23-0451-4000-b000-000000000000', 'handle': '0x0031'}}}, 'plugin': {'value': 'cc2650async', 'type': 'string', 'description': 'Sensortag CC2650 async type plugin', 'default': 'cc2650async'}, 'is_connected': True, 'connectionTimeout': {'value': '10', 'type': 'integer', 'description': 'BLE South Device timeout value in seconds', 'default': '10'}} #012 new config {'shutdownThreshold': {'value': '10', 'type': 'integer', 'description': 'Time in seconds allowed for shutdown to complete the pending tasks', 'default': '10'}, 'plugin': {'value': 'cc2650async', 'type': 'string', 'description': 'Sensortag CC2650 async type plugin', 'default': 'cc2650async'}, 'bluetoothAddress': {'value': '24:71:89:CF:AE:05', 'type': 'string', 'description': 'Bluetooth MAC address', 'default': '24:71:89:CF:AE:05'}, 'connectionTimeout': {'value': '20', 'type': 'integer', 'description': 'BLE South Device timeout value in seconds', 'default': '10'}}
Jan 24 13:15:53 nerd-034 FogLAMP[4897] ERROR: sensortag_cc2650: foglamp.plugins.south.common.sensortag_cc2650: SensorTagCC2650 24:71:89:CF:AE:05 connection failure. Timeout=20 seconds.#012Traceback (most recent call last):#012  File "/home/foglamp/.local/lib/python3.5/site-packages/pexpect/expect.py", line 97, in expect_loop#012    incoming = spawn.read_nonblocking(spawn.maxread, timeout)#012  File "/home/foglamp/.local/lib/python3.5/site-packages/pexpect/pty_spawn.py", line 452, in read_nonblocking#012    raise TIMEOUT('Timeout exceeded.')#012pexpect.exceptions.TIMEOUT: Timeout exceeded.#012#012During handling of the above exception, another exception occurred:#012#012Traceback (most recent call last):#012  File "/home/foglamp/Development/FogLAMP/python/foglamp/plugins/south/common/sensortag_cc2650.py", line 151, in __init__#012    self.con.expect('.*Connection successful.*\[LE\]>', timeout=int(timeout))#012  File "/home/foglamp/.local/lib/python3.5/site-packages/pexpect/spawnbase.py", line 315, in expect#012    timeout, searchwindowsize, async)#012  File "/home/foglamp/.local/lib/python3.5/site-packages/pexpect/spawnbase.py", line 339, in expect_list#012    return exp.expect_loop(timeout)#012  File "/home/foglamp/.local/lib/python3.5/site-packages/pexpect/expect.py", line 104, in expect_loop#012    return self.timeout(e)#012  File "/home/foglamp/.local/lib/python3.5/site-packages/pexpect/expect.py", line 68, in timeout#012    raise TIMEOUT(msg)#012pexpect.exceptions.TIMEOUT: Timeout exceeded.#012<pexpect.pty_spawn.spawn object at 0x7f14870df898>#012command: /usr/bin/gatttool#012args: ['/usr/bin/gatttool', '-b', '24:71:89:CF:AE:05', '--interactive']#012searcher: None#012buffer (last 100 chars): b'4:71:89:CF:AE:05\r\n\x1b[0;91mError: \x1b[0mconnect: Device or resource busy (16)\r\n[24:71:89:CF:AE:05][LE]> '#012before (last 100 chars): b'4:71:89:CF:AE:05\r\n\x1b[0;91mError: \x1b[0mconnect: Device or resource busy (16)\r\n[24:71:89:CF:AE:05][LE]> '#012after: <class 'pexpect.exceptions.TIMEOUT'>#012match: None#012match_index: None#012exitstatus: None#012flag_eof: False#012pid: 5009#012child_fd: 22#012closed: False#012timeout: 30#012delimiter: <class 'pexpect.exceptions.EOF'>#012logfile: None#012logfile_read: None#012logfile_send: None#012maxread: 2000#012ignorecase: False#012searchwindowsize: None#012delaybeforesend: 0.05#012delayafterclose: 0.1#012delayafterterminate: 0.1
```

**Comment by praveen-garg** [comment]
@ashish-jabble  I think Poll plugin is shutting down properly and dis-connected. it just did not get connected to (re)start
> Old config for CC2650POLL plugin .... 
logs before this line suggest that. and this line appears on when (re)start with new (change notified) config.

**Comment by praveen-garg** [comment]
re: http south

because a task is assigned to sever `server = asyncio.ensure_future(coro)` in plugin_start, please open a bug for this.

File: python/foglamp/services/core/server.py (hunk 1 above)
Comment (MarkRiddoch) [reply python/foglamp/services/core/server.py]: Yes, we can remove that code

**Pull request merged**